qiskit/providers/aer/jobs/utils.py

```diff
@@ -55,35 +56,139 @@ def wrapper(*args, **kw):
     return wrapper
 
 
-def split_qobj(qobj, max_size=None, qobj_id=None):
-    """Split a qobj and return a list of qobjs each with a single experiment.
+def _copy_qobj_for_noise(qobj, max_shot_size, qobj_id):
 
-    Args:
-        qobj (Qobj): The input qobj object to split
-        max_size (int or None): the maximum number of circuits per job. If
-            None don't split (Default: None).
-        qobj_id (str): Optional, set a fixed qobj ID for all subjob qobjs.
+    num_shot_jobs, shot_mod = divmod(qobj.config.shots, max_shot_size)
+    qobj_list = []
 
-    Returns:
-        List: A list of qobjs.
-    """
+    if shot_mod == 0 and num_shot_jobs == 1:
+        return qobj
+
+    if shot_mod > 0:
+        setattr(qobj.config, "shots", shot_mod)
+        for experiment in qobj.experiments:
+            _id = str(uuid.uuid4())
+            setattr(experiment.header, "metadata", {"id": _id})
+        qobj_list.append(qobj)
+
+    if num_shot_jobs > 1:
+        _qid = qobj_id or str(uuid.uuid4())
+        _config = copy.deepcopy(qobj.config)
+        setattr(_config, "shots", max_shot_size)
+        experiment_list = []
+        for experiment in qobj.experiments:
+            _id = str(uuid.uuid4())
+            for _ in range(num_shot_jobs):
+                cpy_exp = copy.deepcopy(experiment)
+                setattr(cpy_exp.header, "metadata", {"id": _id})
+                experiment_list.append(cpy_exp)
+        qobj_list.append(QasmQobj(_qid, _config, experiment_list, qobj.header))
+
+    return qobj_list
+
+
+def _split_qobj(qobj, max_size, qobj_id, seed):
     # Check if we don't need to split
     if max_size is None or not max_size > 0:
-        return qobj
+        return qobj, seed
+
     num_jobs = ceil(len(qobj.experiments) / max_size)
     if num_jobs == 1:
-        return qobj
+        return qobj, seed
 
+    qobjs = []
     # Check for parameterizations
     params = getattr(qobj.config, 'parameterizations', None)
-    qobjs = []
+
+    exp_id = None
+    shift_index = 0
+    seed_shift = 256
```

Comment:
Can this be made same as in C++ so RNG is same for manually splitting and running using:
`run(k * [circuit], shots=shots/k)` as with doing this automatic splitting.

Reply:
I moved the operation to set seed and enable it as the c++ operation.